### PR TITLE
Sched 96 create workspace roles

### PR DESCRIPTION
**Pull request by Cteklitz**

### CODE REVIEW

File: server/api/serializers/role.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from rest_framework import serializers
+from ..models import WorkspaceRole, MemberRole
+
+class RoleSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = WorkspaceRole
+        fields = ['name','pay_rate'] # workspace not included in serializer because it would need to be called 'workspace' rather than 'workspace_id', which would be inconsitent with other requests
```
Comment (kinneyan) [inline server/api/serializers/role.py:7]: Can this be changed now that #20 has been merged?

File: server/api/tests/role.py [hunk 2]
```diff
@@ -0,0 +1,802 @@
...
+        self.client.force_authenticate(user=self.user2) # change to send request from user 2
+
+        response = self.client.put(self.url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_create_workspace_role_as_non_member(self):
+        data = {'workspace_id': self.workspace.id, 'name': 'test name1', 'pay_rate': 15.00}
+
+        self.client.force_authenticate(user=self.user3) # change to send request from user 2
```
Comment (kinneyan) [inline server/api/tests/role.py:101]: Does this need to be changed?

File: server/api/tests/role.py
Comment (kinneyan) [inline server/api/tests/role.py:1]: What a beast.

File: server/api/views/workspace.py
Comment (kinneyan) [inline server/api/views/workspace.py:1]: Mistyping "exist" twice within 5 lines is pretty funny.

**Review by kinneyan — APPROVED** [review]
Looks good. Nice work.

File: server/api/serializers/role.py (hunk 1 above)
Comment (Cteklitz) [reply server/api/serializers/role.py:7]: this cant be changed due to #20 being merged, since we use '_id' for request param names but not in the db.

Comment (kinneyan) [reply server/api/serializers/role.py:7]: Ah I understand now. Makes sense. I'm sure there is some way around it but as long as it works correctly and efficiently as-is, it is fine.

File: server/api/tests/role.py (hunk 2 above)
Comment (Cteklitz) [reply server/api/tests/role.py:101]: creating workspace roles still needs workspace_id provided if thats what you meant

Comment (Cteklitz) [reply server/api/tests/role.py:101]: wait you mean the auth; The auth is right, the comment is wrong

**Pull request merged**